tests/core/test_datamodules.py

```diff
@@ -13,6 +16,66 @@ def test_base_datamodule(tmpdir):
     dm.setup()
 
 
+def test_base_datamodule_with_verbose_setup(tmpdir):
+    dm = TrialMNISTDataModule()
+    dm.prepare_data()
+    dm.setup('fit')
+    dm.setup('test')
+
+
+def test_data_hooks_called(tmpdir):
+    dm = TrialMNISTDataModule()
+    assert dm.has_prepared_data is False
+    assert dm.has_setup_fit is False
+    assert dm.has_setup_test is False
+
+    dm.prepare_data()
+    assert dm.has_prepared_data is True
+    assert dm.has_setup_fit is False
+    assert dm.has_setup_test is False
+
+    dm.setup()
+    assert dm.has_prepared_data is True
+    assert dm.has_setup_fit is True
+    assert dm.has_setup_test is True
+
+
+def test_data_hooks_called_verbose(tmpdir):
+    dm = TrialMNISTDataModule()
+    assert dm.has_prepared_data is False
+    assert dm.has_setup_fit is False
+    assert dm.has_setup_test is False
+
+    dm.prepare_data()
+    assert dm.has_prepared_data is True
+    assert dm.has_setup_fit is False
+    assert dm.has_setup_test is False
+
+    dm.setup('fit')
+    assert dm.has_prepared_data is True
+    assert dm.has_setup_fit is True
+    assert dm.has_setup_test is False
+
+    dm.setup('test')
+    assert dm.has_prepared_data is True
+    assert dm.has_setup_fit is True
```

Comment:
```suggestion
```
I think, during testing phase, it should be concerned with `has_setup_test` only.

Reply:
I was trying to be super clear that `dm.has_setup_fit` doesn't accidentally get changed to `False` when you call `dm.setup('test')` after calling `dm.setup('fit')`. 

If that makes sense to you, I'll leave it. If you still disagree, let me know and I'll take it out.

Reply:
Good point. It should not be changed to `False`. But still, during testing, it's of no use I guess. Will be changed to `True` when we do a `.fit` again. I would suggest keeping it for now, but maybe try `assert dm.has_setup_fit is False` just before the final merge once it's ready and see if it fails. Also maybe change `assert dm.has_setup_fit is True` to `assert dm.has_setup_fit` and `assert dm.has_setup_fit is False` to `assert not dm.has_setup_fit`?